src/containers/SearchPage/SearchPage.shared.js

```diff
@@ -495,3 +533,24 @@ export const getDatesAndSeatsMaybe = (currentParams, newParams) => {
       : { seats: null, dates: null };
   return datesAndSeatsMaybe;
 };
+
+/**
+ * Fetch an array of parameters for createResourceLocatorString function based on whether the
+ * target search page has a listingType route parameter
+ * @param {*} routeConfiguration
+ * @param {*} search
+ * @param {*} listingType
+ * @returns Array:
+ * [
+ *    routeName – either 'SearchPageWithListingType' or 'SearchPage',
+ *    routeConfiguration,
+ *    pathParams – either { listingType } or {},
+ *    search,
+ *
+ * ]
+ */
+export const getResourceLocatorStringParams = (routeConfiguration, search, listingType) => {
+  const routeName = listingType ? 'SearchPageWithListingType' : 'SearchPage';
+  const pathParams = listingType ? { listingType } : {};
+  return [routeName, routeConfiguration, pathParams, search];
+};
```

Comment:
Perhaps you could go to this kind of direction
```suggestion
// Note: pathName could be picked from React Router's location object
// const location = useLocation();
// location?.pathname
export const getResourceLocatorStringParams = (routeConfiguration, pathParams, search, pathName) => {
      const testURL = new URL(`http://my.marketplace.com${pathname}`);
      const matchedRoutes = matchPathname(testURL.pathname, routeConfiguration);
      const routeName = matchedRoutes?.[0]?.found?.route?.name || 'SearchPage';

      return [routeName, routeConfiguration, pathParams, search];
};
```

Reply:
Updated the function to take routeConfiguration and location as params and to return routeName and pathParams fetched with matchedRoutes!